### PR TITLE
Basic Asyncio Connection adapter

### PR DESCRIPTION
**Pull request by mosquito**
That's simple adapter for asyncio. It's works fine for me. I already tried to use #694 but it's doesn't works.

### CODE REVIEW

**Comment by vitaly-krugl** [comment]
Hi Mosquito, thank you for submitting this PR. I took a brief look, and a few things jumped out right away that are lacking:
# Documentation (helpful docstrings)
# Usage examples
# Tests

**Comment by vitaly-krugl** [comment]
I would recommend starting with helpful docstrings and usage examples that would help us evaluate the API. Once the API is reviewed/refined, then tests would be the next step before the PR may be considered for merging.

Best regards,
Vitaly

**Comment by mosquito** [comment]
> Hi Mosquito, thank you for submitting this PR. I took a brief look, and a few things jumped out right away that are lacking:
> 1. Documentation (helpful docstrings)
> 2. Usage examples
> 3. Tests

Done 😊

**Comment by vitaly-krugl** [comment]
@mosquito, it looks like the resulting adapter instance would be API-compatible with TornadoConnection and SelectConnection adapters.

This would make it super easy to plug it into the async adapter acceptance test framework simply by adding a tiny function similar to `tornado_test` in https://github.com/pika/pika/blob/7d0ef0bb9db4aa9f9e7020e9539c171c4ce5c45d/tests/acceptance/async_test_base.py#L191.

Yours will need to have a `skip` decorator so that it only runs on versions of python 3.x, where the compatible asyncio is available.

File: pika/adapters/asyncio_connection.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    :param on_open_callback: The method to call when the connection is open
+    :type on_open_callback: method
+    :param on_open_error_callback: Method to call if the connection cant be opened
+    :type on_open_error_callback: method
+    :param asyncio.AbstractEventLoop loop: By default asyncio.get_event_loop()
+
+    """
+    def __init__(self, parameters=None, on_open_callback=None,
+                 on_open_error_callback=None,
+                 on_close_callback=None, loop=None):
```
Comment (vitaly-krugl) [inline pika/adapters/asyncio_connection.py]: @mosquito, please model the constructor args on TornadoConnection/SelectConnection for consistency. This should also facilitate easy integration into the async adapter acceptance test framework. See 

* https://github.com/pika/pika/blob/43936088e1183ec5902f2ba2992c902e87f5460b/pika/adapters/tornado_connection.py#L30-L35
* https://github.com/pika/pika/blob/43936088e1183ec5902f2ba2992c902e87f5460b/pika/adapters/select_connection.py#L68-L73

**Comment by vitaly-krugl** [comment]
@gmr, I am not at all familiar with asyncio at this point. I am counting on you for evaluating the suitability of this adaptation to the `asyncio` model. Thx.

**Comment by vitaly-krugl** [comment]
Thanks, @mosquito. Integrating into the acceptance test framework gives us a better idea where the breaking points might be. Presently, asyncio is failing some tests on python 3.4+.

I also see a bunch of new `ResourceWarning` messages at the end of the logs. They indicate that there are new resource leaks. This is something to watch out for. Perhaps fixing the broken functionality will resolve the leaks as well. If not, then the leaks need to be fixed. You will want to compare to other recent PR builds.

**Comment by lukebakken** [comment]
Closing this PR as your commits are included in #805 - thank you!

**Pull request closed without merging**